### DISCUSSION TITLE
Clean up of IntersectionObserver on Component Did Unmount

### DISCUSSION THREAD

**Discussion by andrewcorliss** [thebuilder/react-intersection-observer#569] [Q&A] [answered]
Hello,
I am looking over the documentation and reviewing the code and I am curious if the Observers are cleaned up after navigating away from the components. I am working with an application where I need to keep the observer active for the entire lifecycle and I am thinking it is causing some issues with memory.

Is there a way to manually clean up after the component unmounts?

Thank you

**Comment by thebuilder** [comment, marked as answer]
Hey @andrewcorliss 

Yes, the Observers are destroyed if they no longer have any active references (from `useInView` or `<InView>`), but they are also reused if an observer already exists with the same options.

If you need to have control over when the `unmount()` happens, then you can use the `observe()` method directly:
https://github.com/thebuilder/react-intersection-observer#low-level-api

Keep in mind, it will still destroy the observers if they are no longer used after you call `unmount`, and also that you should call `unmount` whenever your component is destroyed. So not sure if you'd gain anything from it?

Are you sure it's causing issues?

**Reply by andrewcorliss** [reply]
Thanks,
I will try that.
As for how sure I am still trying to debug the exact cause of the issue.  So for now I am trying to reduce any factors that could lead to it.
